tests/unittests/tests-nanocoap/tests-nanocoap.c

```diff
@@ -1059,11 +1059,35 @@ static void test_nanocoap__token_length_over_limit(void)
     TEST_ASSERT_EQUAL_INT(8, coap_get_token_len(&pkt));
     TEST_ASSERT_EQUAL_INT(0, pkt.payload_len);
 
-    /* Invalid packet (TKL = 9) */
+    /* Invalid packet (TKL = 15) */
     res = coap_parse(&pkt, buf_invalid, sizeof(buf_invalid));
     TEST_ASSERT_EQUAL_INT(-EBADMSG, res);
 }
 
+/*
+ * Verifies that coap_parse() recognizes token length bigger than allowed.
+ */
+static void test_nanocoap__token_length_ext(void)
+{
+    const char *token = "0123456789ABCDEF";
+
+    uint8_t buf[32];
+    coap_hdr_t *hdr = (void *)buf;
```

Comment:
Why is it done this way? I don't spot a reason not to do `coap_hdr_t hdr;` and then later on passing `&hdr`?

Reply:
`coap_hdr_t` only fits the non-extended token